Submissions/Ekey_Week 3 - Functions and Objects.playground/Contents.swift

```diff
@@ -0,0 +1,263 @@
+//:
+//class Shape {
+//    var numberOfSides = 0
+//    func simpleDescription() -> String {
+//        return "A shape with \(numberOfSides) sides."
+//    }
+//}
+
+//Laurie:  Answer
+class Shape {
+    var numberOfSides = 0
+    let myVar = "Hello"
+    func simpleDescription() -> String {
+        return "A shape with \(numberOfSides) sides."
+    }
+    func takesArgument(_ greeting: String) {
+        print(greeting + " my friend")
+    }
+    init(count: Int) {
+        numberOfSides = count
+    }
+}
+
+let testLetwithClass = Shape(count: 5)
+testLetwithClass.takesArgument("Hola")
+testLetwithClass.numberOfSides = 6
+// cannot modify let property unless use mutable
+
+
+
+//: 5.)
+//:
+//: Make a subclass of `NamedShape` called `Circle` that takes a radius and a name as arguments to its initializer. Implement an `area()` and a `simpleDescription()` method on the `Circle` class.
+//:
+
+// Amanda -- I would change this to take just the radius, Circle itself should know what it is. i.e. a Circle -- so only 1 parm on the init/constructor
+
+class NamedShape {
+    var numberOfSides: Int = 0
+    var name: String
+    
+    init(name: String) {
+        self.name = name
+    }
+    
+    func simpleDescription() -> String {
+        return "A shape with \(numberOfSides) sides."
+    }
+}
+
+class Circle : NamedShape {
+    var radius: Double = 0
+    
+    // Laurie slight change... each object should know how to handle itself - shouldn't have to pass in "circle" from outside..
+    init( radius: Double)
+    {
+        self.radius = radius
+        super.init(name: "Circle")
+    }
+    
+    // What Amanda asked for...
+//    init(name: String, radius: Double)
+//    {
+//        self.radius = radius
+//        super.init(name: name)
+//    }
+    override func simpleDescription() -> String {
+        let myArea = area()
+        return "A \(name) with \(numberOfSides) sides, a radius of \(radius) and an area of \(myArea)"
+    }
+    
+    /// area of circle = pi r^2
+    func area() -> Double{
+        return Double.pi * (radius * radius)
+    }
+}
+
+var myShape = Circle(/*name: "Circle",*/ radius: 5.0)
+myShape.simpleDescription()
+
+//: ## Enumerations and Structures
+//: 6.)
+//:
+//: Write a function that compares two `Rank` values by comparing their raw values.
+//:
+enum Rank: Int {
+    case ace = 1
+    case two, three, four, five, six, seven, eight, nine, ten
+    case jack, queen, king
+    
+    func simpleDescription() -> String {
+        switch self {
+        case .ace:
+            return "ace"
+        case .jack:
+            return "jack"
+        case .queen:
+            return "queen"
+        case .king:
+            return "king"
+        default:
+            return String(self.rawValue)
+        }
+    }
+}
+
+func compareRank(r1: Rank, r2: Rank) -> Bool
+{
+    return r1.rawValue == r2.rawValue ? true : false
```

Comment:
using a ternary to return true or false is itself a red flag that you should remove the ternary; the result of a == comparison is itself a boolean value, you can simply
`return r1.rawValue == r2.rawValue`

that being said, please use descriptive names over `r1` and `r2`. For someone who likes syntactic sugar you'll be happy to hear that Swift programmers like descriptive variable names, such as `firstRank` and `secondRank`. Autocomplete is your friend.